packages/client/src/components/SafeSettings/components/Assets/AddVault.js

```diff
@@ -1,45 +1,57 @@
 import { useState } from "react";
 import { COIN_TYPE_TO_META } from "constants/maps";
 import Dropdown from "components/Dropdown";
-const AddVault = ({ onCancel, onNext }) => {
+import { useVaults } from "hooks";
+
+const AddVault = ({ address, onCancel, onNext }) => {
   const onNextClick = () => {
     onNext({
       coinType,
     });
   };
+  const vaultProps = useVaults(address);
+
+  const [coinType, setCoinType] = useState("Select Token");
+  const vaultName = COIN_TYPE_TO_META[coinType]?.vaultName;
+
+  const isFormValid =
+    vaultName &&
+    !vaultProps.vaults[address]?.find((id) => id.indexOf(vaultName) >= 0);
+
+  const isDisabled = [
+    isFormValid ? "is-link" : "is-light is-disabled",
+  ];
 
   const coinTypes = Object.entries(COIN_TYPE_TO_META).map((type) => ({
     itemValue: type[0],
     displayText: type[1].displayName,
   }));
 
-  const [coinType, setCoinType] = useState(coinTypes[0].itemValue);
-
   return (
-    <>
-      <div className="p-5">
-        <h2 className="is-size-4 has-text-black">Add Vault</h2>
+    <div className="p-5 has-text-grey">
+      <div className="flex-1 is-flex is-flex-direction-column">
+        <label className="has-text-grey mb-2">Tokens</label>
+        <Dropdown
+          value={coinType}
+          values={coinTypes}
+          setValue={setCoinType}
+          style={{ height: "45px" }}
+        />
+        {vaultName && !isFormValid && (
+          <p className="has-text-red mt-2">
+            This vault has already been added.
+          </p>
+        )}
       </div>
-      <div className="border-light-top p-5 has-text-grey">
-        <div className="flex-1 is-flex is-flex-direction-column">
-          <label className="has-text-grey mb-2">Contract Name</label>
-          <Dropdown
-            value={coinType}
-            values={coinTypes}
-            setValue={setCoinType}
-            style={{ height: "45px" }}
-          />
-        </div>
-        <div className="is-flex is-align-items-center mt-6">
-          <button className="button flex-1 is-border mr-2" onClick={onCancel}>
-            Cancel
-          </button>
-          <button className="button flex-1 is-primary" onClick={onNextClick}>
-            Next
-          </button>
-        </div>
+      <div className="is-flex is-align-items-center mt-6">
+        <button className="button flex-1 is-border mr-2" onClick={onCancel}>
+          Cancel
+        </button>
+        <button className={`button flex-1 is-primary ${isDisabled}`} onClick={onNextClick}>
```

Comment:
```suggestion
        <button className={`button flex-1 is-primary ${isDisabled}`} disabled={!isFormValid} onClick={onNextClick}>
```

Reply:
Is it okay if we go with the **pointer-events: none** for every button that has a class disabled?
I added this to App.sass, since I noticed that we have this class appended in a couple of places, but the button wasn't actually disabled -> so this will work for all of them

Reply:
I think we should use the "disabled" class purely for styling purpose and use semantic HTML syntax. Bulma also deprecated the "is-disabled" class in favor of html disabled tag (https://bulma.io/documentation/elements/button/)

Reply:
Updated!